### ISSUE TITLE
New command: release

### ISSUE THREAD

**Issue by miketheman** [crystal-lang/shards#98]
When developing a library, it is convenient to have a method of releasing a version with well-known semantics, such as version tagging and github push.

In Ruby's Bundler, it is common to use `rake release` which performs a few actions, including a push to the remote rubygems.org site.

While Crystal Shards don't yet have a "release" location, a command like `shards release` from a working directory of a shard could validate that the current state is safe, tag the version, and push to github.

**Comment by duraki** [comment]
Any update on this?

**Comment by wontruefree** [comment]
`rake release` mostly just tags and pushes to rubygems.com right?
would we want shards release to do anything more?

**Comment by miketheman** [comment]
@wontruefree I'd take that as a starting point. I think it does a "git push --tags" as well - to ensure that the tag actually makes it to the remote.

**Comment by straight-shoota** [comment]
I don't think this makes any sense as a shards subcommand. `shards` shouldn't care about how you release your shard. There is no one-size-fits-all solution.

It is much easier to use a customized Makefile or CI script for a release process.
*Maybe* #202 could provide a way to integrate a custom release script into `shards` CLI.

**Comment by caspiano** [comment]
Having a `shards release` command makes sense to me.
As `shards` uses semver and the reference is pulled from version control tags, it's a pretty limited surface to manage... but the operations happen enough that it seems routine for me, and worthy of a tool enhancement.

Something like `shards release <major|minor|patch>` that updates the `shard.yml` and attempts to tag the current commit with the updated version would be handy IMO.

**Comment by straight-shoota** [comment]
The other day I read this introduction blog post for `cargo dist`: https://blog.axo.dev/2023/02/cargo-dist

A basic building block for that is `cargo release` which seems to work quite well. Cargo has a centralized registry where the release can be published to with `cargo publish`. That Crystal doesn't have, but publishing would just mean to push the tag to the public repository. So maybe this could be available as a convenience feature in a single step as well.
